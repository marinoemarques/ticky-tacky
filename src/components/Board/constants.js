export const INITIAL_STATE = {
  moves: []
}

export const SQUARE_PLAYED = 'SQUARE_PLAYED'
