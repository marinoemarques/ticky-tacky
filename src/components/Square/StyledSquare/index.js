import styled from 'styled-components'

import {
  PLAYER_ONE,
  TACKY_COLOUR,
  TICKY_COLOUR
} from '../../../constants'

function getColour ({ isLoser, player }) {
  if (isLoser) { return 'Gainsboro' }

  return player === PLAYER_ONE
    ? TICKY_COLOUR
    : TACKY_COLOUR
}

const StyledSquare = styled.span`
  align-self: stretch;
  background-color: White;
  color: ${getColour};;
  cursor: ${props => props.onClick ? 'pointer' : 'not-allowed'};
  font-size: 16vh;
  font-weight: bold;
  grid-area: ${props => props.area || '0'};
  justify-self: stretch;
  text-align: center;
  text-transform: uppercase;
`
StyledSquare.displayName = 'StyledSquare'

export default StyledSquare
