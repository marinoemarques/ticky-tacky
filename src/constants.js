import { length, pipe, values } from 'ramda'

const size = pipe(values, length)

// State
export const INITIAL_STATE = {
  moves: []
}

// Actions
export const SQUARE_PLAYED = 'SQUARE_PLAYED'

// Players
export const PLAYER_ONE = 'x'
export const PLAYER_TWO = 'o'
export const UNPLAYED = ''

// Colours
export const TACKY_COLOUR = 'SlateBlue'
export const TICKY_COLOUR = 'Crimson'

// Squares
export const SQUARES = {
  topLeft: 0,
  topCenter: 1,
  topRight: 2,
  middleLeft: 3,
  middleCenter: 4,
  middleRight: 5,
  bottomLeft: 6,
  bottomCenter: 7,
  bottomRight: 8
}
export const NUMBER_OF_SQUARES = size(SQUARES)

// Game play
export const SQUARES_TO_WIN = 3
export const WINNING_PATTERNS = [
  [
    SQUARES.topLeft,
    SQUARES.topCenter,
    SQUARES.topRight
  ],
  [
    SQUARES.middleLeft,
    SQUARES.middleCenter,
    SQUARES.middleRight
  ],
  [
    SQUARES.bottomLeft,
    SQUARES.bottomCenter,
    SQUARES.bottomRight
  ],
  [
    SQUARES.topLeft,
    SQUARES.middleLeft,
    SQUARES.bottomLeft
  ],
  [
    SQUARES.topCenter,
    SQUARES.middleCenter,
    SQUARES.bottomCenter
  ],
  [
    SQUARES.topRight,
    SQUARES.middleRight,
    SQUARES.bottomRight
  ],
  [
    SQUARES.topLeft,
    SQUARES.middleCenter,
    SQUARES.bottomRight
  ],
  [
    SQUARES.topRight,
    SQUARES.middleCenter,
    SQUARES.bottomLeft
  ]
]
