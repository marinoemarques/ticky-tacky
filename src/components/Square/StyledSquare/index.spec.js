import * as React from 'react'

import { shallow } from 'enzyme'

import { PLAYER_ONE, PLAYER_TWO } from '../../../constants'

import StyledSquare from './'

describe('StyledSquare', () => {
  it('matches the snapshot with no player', () => {
    expect(toJson(shallow(
      <StyledSquare />
    ).first().render())).toMatchSnapshot()
  })

  it(`matches the snapshot with player ${PLAYER_ONE}`, () => {
    expect(toJson(shallow(
      <StyledSquare
        area='4'
        player={PLAYER_ONE}
      >{PLAYER_ONE}</StyledSquare>
    ).first().render())).toMatchSnapshot()
  })

  it(`matches the snapshot with player ${PLAYER_TWO}`, () => {
    expect(toJson(shallow(
      <StyledSquare
        area='0'
        player={PLAYER_TWO}
      >{PLAYER_TWO}</StyledSquare>
    ).first().render())).toMatchSnapshot()
  })
})
