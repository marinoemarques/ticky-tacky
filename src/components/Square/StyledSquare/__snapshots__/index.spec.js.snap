// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledSquare matches the snapshot with no player 1`] = `
.c0 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: White;
  color: SlateBlue;
  cursor: not-allowed;
  font-size: 16vh;
  font-weight: bold;
  grid-area: 0;
  justify-self: stretch;
  text-align: center;
  text-transform: uppercase;
}

<span
  class="c0"
/>
`;

exports[`StyledSquare matches the snapshot with player o 1`] = `
.c0 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: White;
  color: SlateBlue;
  cursor: not-allowed;
  font-size: 16vh;
  font-weight: bold;
  grid-area: 0;
  justify-self: stretch;
  text-align: center;
  text-transform: uppercase;
}

<span
  class="c0"
>
  o
</span>
`;

exports[`StyledSquare matches the snapshot with player x 1`] = `
.c0 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  background-color: White;
  color: Crimson;
  cursor: not-allowed;
  font-size: 16vh;
  font-weight: bold;
  grid-area: 4;
  justify-self: stretch;
  text-align: center;
  text-transform: uppercase;
}

<span
  class="c0"
>
  x
</span>
`;
